lib/plug/static.ex

```diff
@@ -145,6 +149,14 @@ defmodule Plug.Static do
   defp allowed?(nil, _list),  do: true
   defp allowed?(only, [h|_]), do: h in only
 
+  defp prefix_allowed?(_only, []), do: false
+  defp prefix_allowed?(nil, _list), do: true
+  defp prefix_allowed?(only, [h|_]) do
+    Enum.any?(only, fn pattern ->
+      :binary.match(h, pattern) != :nomatch
```

Comment:
The reason I don't like prefix_only is because it will lead to a large
surface API, specially if we want to add more matches in the future (say
suffix matches). Implementing it would be a matter of calling
Enum.partition, no?

On Tuesday, March 1, 2016, Eric Meadows-Jönsson <notifications@github.com>
wrote:

> In lib/plug/static.ex
> <https://github.com/elixir-lang/plug/pull/359#discussion_r54568616>:
>
> > @@ -145,6 +149,14 @@ defmodule Plug.Static do
> >    defp allowed?(nil, _list),  do: true
> >    defp allowed?(only, [h|_]), do: h in only
> >
> > +  defp prefix_allowed?(_only, []), do: false
> > +  defp prefix_allowed?(nil, _list), do: true
> > +  defp prefix_allowed?(only, [h|_]) do
> > +    Enum.any?(only, fn pattern ->
> > +      :binary.match(h, pattern) != :nomatch
>
> You're right.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elixir-lang/plug/pull/359/files#r54568616>.
>


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*